### DISCUSSION TITLE
Idena Desktop App validation troubleshooting

### DISCUSSION THREAD

**Discussion by rioda-org** [idena-network/idena-go#799] [General]
## 👋 Let's make a guide for members who use Desktop App, so they can troubleshoot their validation
[Web App troubleshooting is here](https://github.com/idena-network/idena-web/discussions/123)

## General suggestions when using Desktop App:
- [Check your internet connection quality here](http://www.dslreports.com/speedtest). If your connection is graded A or B, you have good connection. If it's C or worse, it is bad. If you are connected via Wifi, try connecting using LAN cable and check connection again. If your connection is still bad, you MUST use [Web App](https://app.idena.io) for validation.
- If you are using Windows OS, I recomend you to install free software **NetTime** which will keep your clock in perfect syncronisation which is important for validation (http://www.timesynctool.com/NetTimeSetup-314.exe)
- Make sure you turn off any other aplication that uses internet.

Let us start with troubleshooting this log file and I will try to form some guide based on that.
[output.log](https://github.com/idena-network/idena-go/files/7188422/output.log)

This member got result "Late submision". If we check out his flip for this validation, we can see that Challenged identities did provide answers for it, so he did not have problem of sending decryption keys
https://scan.idena.io/flip/bafkreie5ljsllvhq4kijbokstp7bywp3psrq6o7rp3z2znywhlirbeaaxy

Next we need to see when did his node send `Short answers proof` tx? We search by `type=5` to find it.
```shell
INFO [09-01|13:31:30.019] Broadcast ceremony tx                    type=5 hash=0xbca80c95478c448b4515b091b56d743fcc4902ce10f9bb9bf6f6787d51fb5b55
```

We see that the time is under short session which is good, but we need to see bellow to how many peers did he send this transaction

```shell
INFO [09-01|13:31:30.019] short answers submitting response 
INFO [09-01|13:31:30.077] Sent high priority msg, code 14          id=QmbdR8C8GEWQXeSoUUrUv51LyKGLP2eGpd7zawSpbiLxeD
```
Here we can see that his node had sent this transaction only to one peer `id=Qmbd...`. That means that his node was connected only to one peer at that time. When user have good connection, you will see multiple `Sent high priority msg, code 14` all one after another.

From that I presume that this transaction did not reach more than 50% nodes in time so it has result Late submission.

His PC is connected using Wifi and his connection is rated as bad:
http://www.dslreports.com/speedtest/69544139
Very bad, I have suggested to connect using lan cable.

**Comment by hlolve** [comment]
As you said one peer, and with problems

 ```shell
ERROR[09-01|13:31:45.598] error while writing to stream            id=QmbdR8C8GEWQXeSoUUrUv51LyKGLP2eGpd7zawSpbiLxeD err="closed stream"
INFO [09-01|13:31:45.598] Peer disconnected                        id=QmbdR8C8GEWQXeSoUUrUv51LyKGLP2eGpd7zawSpbiLxeD
 ```

This other peer also is there but with problems

 ```shell
INFO [09-01|13:31:43.754] Sent high priority msg, code 14          id=QmYCvtXcVyCcEBoyw1nso7jk4pgfKqjH4BemgasX91NdJH
 ```

 ```shell
ERROR[09-01|13:32:43.756] TIMEOUT while writing to stream          id=QmYCvtXcVyCcEBoyw1nso7jk4pgfKqjH4BemgasX91NdJH addr=/ip4/149.28.166.78/tcp/40405
INFO [09-01|13:32:43.756] Peer disconnected                        id=QmYCvtXcVyCcEBoyw1nso7jk4pgfKqjH4BemgasX91NdJH
 ```

So, the problem is his connection

**Reply by rioda-org** [reply]
Yes, he used wifi, I have strongly suggested to get lan cable, he agreed :D

**Comment by hlolve** [comment]
The msg sent to 4 peers in total


 ```shell
INFO [09-01|13:31:30.077] Sent high priority msg, code 14          id=QmbdR8C8GEWQXeSoUUrUv51LyKGLP2eGpd7zawSpbiLxeD

INFO [09-01|13:31:31.811] Sent high priority msg, code 14          id=QmSvLbDwYF5JEHD9zvbCgSRTtjGxFfbX2dFrsToX3gz8PT

INFO [09-01|13:31:38.429] Sent high priority msg, code 14          id=QmNnz9nYBqUXy89Fh9k6tCfmm7b1xx8n992xdsAbwXECTq

INFO [09-01|13:31:43.754] Sent high priority msg, code 14          id=QmYCvtXcVyCcEBoyw1nso7jk4pgfKqjH4BemgasX91NdJH
 ```

 ```shell

ERROR[09-01|13:31:45.598] error while writing to stream            id=QmbdR8C8GEWQXeSoUUrUv51LyKGLP2eGpd7zawSpbiLxeD err="closed stream"
INFO [09-01|13:31:45.598] Peer disconnected                        id=QmbdR8C8GEWQXeSoUUrUv51LyKGLP2eGpd7zawSpbiLxeD



ERROR[09-01|13:32:13.385] error while writing to stream            id=QmSvLbDwYF5JEHD9zvbCgSRTtjGxFfbX2dFrsToX3gz8PT err="session shut down"
INFO [09-01|13:32:13.385] Peer disconnected                        id=QmSvLbDwYF5JEHD9zvbCgSRTtjGxFfbX2dFrsToX3gz8PT


WARN [09-01|13:41:15.933] process batch - timeout was reached      component=downloader peer=QmNnz9nYBqUXy89Fh9k6tCfmm7b1xx8n992xdsAbwXECTq
INFO [09-01|13:41:15.933] Start process batch                      component=downloader from=3322964 to=3322970

ERROR[09-01|13:32:43.756] TIMEOUT while writing to stream          id=QmYCvtXcVyCcEBoyw1nso7jk4pgfKqjH4BemgasX91NdJH addr=/ip4/149.28.166.78/tcp/40405
INFO [09-01|13:32:43.756] Peer disconnected                        id=QmYCvtXcVyCcEBoyw1nso7jk4pgfKqjH4BemgasX91NdJH

 ```

Just 1 **QmNnz9nYBqUXy89Fh9k6tCfmm7b1xx8n992xdsAbwXECTq** was good at that time and you can see problems again at
INFO [09-01|13:41:15.933] Start process batch

**Reply by hlolve** [reply]
When the connection from node owner side is OK you will see in node log, all the "Sent high priority msg, code 14" sent to all connected peers at the same time.

**Comment by hlolve** [comment]
If you are validating with desktop app, running local node, do not stop app until results are ready (does not matter if node is mining or not)

**Reply by rioda-org** [reply]
Why is that important?

**Reply by hlolve** [reply]
In today's validation, sent messages with answers were not included in blocks during validation (all empty), only after long session finished, started to include txs, so if someone think after finish send long session answer can stop the app, in this case, the answers maybe not reach the network

**Comment by hogwashpz** [comment]
Hi, 
I noticed that network didn't register one of my short validation answer, I set answers on all flips and submitted ~30 sec before timer runs out.
https://scan.idena.io/identity/0x7e9e8f594236bb4c55441e5d4ce7578ce4354ec5/epoch/75/validation#shortAnswers

I am 100% sure that I clicked on all 6 short session flips, so 100% answers should be there.

Setup: Desktop App + DO VPS (2GB RAM, 2x CPUs, Ubuntu 18.04,  and rebooted 1h before validaiton).

Worst part is when I you look at 2nd Challenge flip, both challenged identities have No answer (first is mine identity), both of challenged have answers on all other flips, so mine is not the single edge case here.
https://scan.idena.io/flip/bafkreigcxpfnjtk7qmlh6zmt43fskt6riixdximgjwplsg2akegnt37gny

```shell
INFO [09-21|13:31:06.670] Broadcast ceremony tx                    type=5 hash=0xa3a3bb708adb7b9d3eee44c80ec495cf4a7e858c1e27bd102f4e8086f679a350
INFO [09-21|13:31:06.670] short answers submitting response 
INFO [09-21|13:31:06.738] Sent own tx push                         hash=0xa3a3bb708adb7b9d3eee44c80ec495cf4a7e858c1e27bd102f4e8086f679a350
INFO [09-21|13:31:06.739] Sent high priority msg, code 14          id=QmYchE6Tr7ZV6K86NaQhn2RFoH9zxkGmsxm7s3xYTC1HkR
INFO [09-21|13:31:06.746] Sent high priority msg, code 14          id=QmdZUwnFkinY3eMVTs22VQgcbDq91TBkMSi1EDXoV2mAmH
INFO [09-21|13:31:06.746] Sent high priority msg, code 14          id=QmYvXPvvNpPD2uL9NSDqbrLr8AYLfXScb6cTFt5Usda1Gm
INFO [09-21|13:31:06.746] Sent high priority msg, code 14          id=QmbaoQVbKjpo9jrSfTiq7aJBnjLgHn7LzdjBvC8BC3yfdd
INFO [09-21|13:31:06.747] Sent high priority msg, code 14          id=QmYoFKcQCaqdjeYmiXxymd5hMBT4VSuyteJgTNpu3Dm6kY
INFO [09-21|13:31:06.747] Sent high priority msg, code 14          id=QmaFfukJCs18xbcZwaA3e3cfXgYsTcfsffmhQwgL2fLJuB
INFO [09-21|13:31:06.748] Sent high priority msg, code 14          id=QmSFGVwuNbBHgNt9cyaApdXK9Pcsj8TwzN72JPCXpVnR8v
INFO [09-21|13:31:06.748] Sent high priority msg, code 14          id=QmY2oGKStpG9Hg4sLtSYvscnCTcNw95NxuKH181ceSG9Jo
INFO [09-21|13:31:06.748] Sent high priority msg, code 14          id=QmYKgkqL9q5CE6Ge6UX7tafG5cwNsVuuSuVN2PvJqddYAd
INFO [09-21|13:31:06.748] Sent high priority msg, code 14          id=QmbQ7eME4mwpCqtzHbyBFuorFdCHKZ9QmK1bTbYaUYsX9f
INFO [09-21|13:31:06.749] Sent high priority msg, code 14          id=QmTTTL9p8ugVsQDviCyBRTXoW5qPtziGoWWZENFSgeEqRk
INFO [09-21|13:31:06.749] Sent high priority msg, code 14          id=QmYNYumkYAWw7G93HqWH27H9EvUHoqsemPDUvo24zp37yJ
INFO [09-21|13:31:06.801] long hashes request 
```

**Reply by rioda-org** [reply]
Nice, can you please attach full validation logs node + client so devs can investigate more. Thanks hog!

**Reply by hogwashpz** [reply]
Here is whole log, including entries during validation.
https://1drv.ms/u/s!AndF8XwhwzLZpBS2yBzt7bCc5eZv?e=FhrCus

**Reply by hlolve** [reply]
Your node restarted 4 times during last validation

**Reply by rioda-org** [reply]
Node version was more power hungry. 0.27.1 has fixed that and hopefully next validation will be smooth.

**Comment by hlolve** [comment]
If you see all flips from that ID seems had problem sending keys https://scan.idena.io/identity/0x8a3A866F45a3fBcE105DFb00B84a440A130d05CD/epoch/75/rewards
